transcom/mymove
# MB-9411 Switch SFTP Connection to use private key refactor

## [MB-9411](https://dp3.atlassian.net/browse/MB-9411)

## Summary

This refactor is actually to make it easier to do the ticket that is linked above and referenced in the title of this PR. The ticket itself is for actually swapping over to use a private key connection for GEX sftp. 
I wanted to do a small refactor first, so that I only need to update the sftp stuff in one place.
The functionality should stay the same, mostly just parts got swapped in for more optional reuse.  

Reuse the cli package to reduce the number of places environment variables are being read, so recycle the usage of the cli package into `syncada_sftp_sender` and `connect_to_gex_via_sftp`.
Adjust `syncada_sftp_sender_test` so that it isn't checking if every single environment variable is available.
Test for the relevant environment variables inside of `gex_sftp_test`.

Unfortunately, the GEX stuff is not able to be tested in exp.

The pull request to actually switch over to key based auth is in #11184.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### How to test

1. Run `make docker_local_ssh_server`
2. Open the CLI to the Docker container you just made
3. Inside the CLI for the Docker container run `cat /etc/ssh/ssh_host_ecdsa_key.pub`
4. Inside of the project's `.envrc.local` set these environment variables, the GEX_SFTP_HOST_KEY should use the key that you got in step 3.

```
export GEX_SFTP_USER_ID=testu
export GEX_SFTP_PASSWORD=testp
export GEX_SFTP_IP_ADDRESS=localhost
export GEX_SFTP_PORT=2222
export SEND_TO_SYNCADA=true
export GEX_SFTP_HOST_KEY='this should be the value of what you retrieved from step 3'
```

Perform the following steps in a separate terminal
5. Run `direnv allow` 
6. Run `make build_tools`
7. Run `milmove-tasks connect-to-gex-via-sftp`

### Backend



[MB-9411]: https://dp3.atlassian.net/browse/MB-9411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
